src/Hng.Web/Controllers/BlogController.cs

```diff
@@ -26,4 +28,38 @@ public async Task<ActionResult<BlogDto>> CreateBlog([FromBody] CreateBlogDto bod
         var response = await _mediator.Send(command);
         return CreatedAtAction(nameof(CreateBlog), response);
     }
+
+    [HttpGet("{id:guid}")]
+    [ProducesResponseType(typeof(BlogDto), StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(FailureResponseDto<string>), StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<BlogDto>> GetBlogById(Guid id)
+    {
+        var query = new GetBlogByIdQuery(id);
+        var response = await _mediator.Send(query);
+
+        return response is null ? NotFound(new FailureResponseDto<BlogDto>
+        {
+            Data = null,
+            Error = "Blog not found",
+            Message = "The requested job does not exist."
+        }) : Ok(response);
+    }
+
+    [HttpGet("")]
+    [ProducesResponseType(typeof(BlogDto), StatusCodes.Status200OK)]
+    public async Task<ActionResult<IEnumerable<BlogDto>>> GetBlogs()
+    {
+        var blogs = await _mediator.Send(new GetBlogsQuery());
+        return Ok(blogs);
```

Comment:
We returned blogs directly here instead of through the success response dto , is this intentional as specified in documentation?